### DISCUSSION TITLE
How to clear the cache folder when closing my application

### DISCUSSION THREAD

**Discussion by azerios** [cefsharp/CefSharp#5012] [General]
Dear,

I'm trying to clear the cache folder when I'm stopping/closing my application.

My code is the following : 
`Cef.ClearCrossOriginWhitelist();`
`Cef.ClearSchemeHandlerFactories();`
`Cef.PreShutdown();`
`Cef.Shutdown();`
`Directory.Delete(_cefsharpCachePath, recursive: true);`

The issue is that one directory is throwing an exception as segmentation_platform and two files : 
ukm_db
ukm_db-journal

How can I stop any process to using them and delete the cache folder ?

Thanks for your help

**Comment by amaitland** [comment]
Converted to discussion as this is not a bug in `CefSharp` directly.

> How can I stop any process to using them and delete the cache folder ?

This is a known issue and will need to be resolved `upstream`.

https://github.com/chromiumembedded/cef/issues/3726

If you absolutely have to delete the folder then you'd likely need a cleanup process that monitors when your main application processes has exited then performs the cleanup.

You can delete cookies, disk cache, etc at runtime.
